tests/Game.class.test.js

```diff
@@ -0,0 +1,965 @@
+/* eslint-disable max-len */
+'use strict';
+
+const Game = require('../src/modules/Game.class');
+
+function sum(game) {
+  let result = 0;
+
+  for (const row of game.getState()) {
+    for (const n of row) {
+      result += n;
+    }
+  }
+
+  return result;
+}
+
+function count(game) {
+  return game.getState().flat().filter(n => n).length;
+}
+
+function copyState(state) {
+  return state.map(row => [...row]);
+}
+
+function transposeState(state) {
+  const result = [];
+
+  for (let col = 0; col < 4; col++) {
+    result[col] = [];
+
+    for (let row = 0; row < 4; row++) {
+      result[col].push(state[row][col]);
+    }
+  }
+
+  return result;
```

Comment:
The `transposeState` function can be optimized by using `Array.prototype.map` and accessing the indices of the original array instead of the nested for-loop.